Server/server.c

```diff
@@ -82,9 +113,48 @@ int main(int argc, char* argv[]) {
 void send_nok() {
 	// TODO
 }
-void ask_areas() {
-// TODO
+
+void send_ok(pid_t sender, int num) {
+	struct message msg_send;
+
+	//init message
+	msg_send.mtype = sender;
+	msg_send.sender = getpid();
+	msg_send.code = OK;
+
+	//send message
+	if (msgsnd(num, &msg_send, sizeof(struct message)-sizeof(long), 0) == -1) {
+		perror("msgsnd");
+		exit(1);
+	}
+	printf("OK message envoyé à %d\n", sender);
+}
+
+void ask_areas(pid_t sender, int num) {
+	char data[MAX_LENGTH_DATA];
+	node_t * temp = areas_list;
+	struct message msg_send;
+
+	//read the list of areas and put it in data
+	while(temp != NULL) {
+		sprintf(data, "%d:", temp->data);
```

Comment:
`sprintf` will overwrite the content, this is a classic Copilot mistake. You should use `strcat()` instead, which requires to do it a little bit differently.
I have done the same thing in Pantalla Deportiva somewhere in the courts part, you should be able to find it if you want to get inspiration